runtime/doc/vim_diff.txt

```diff
@@ -265,6 +273,8 @@ Lua interface (|if_lua.txt|):
 on cancel and completion respectively) via dictionary argument (replaces all 
 other arguments if used).
 
+|input()| and |inputdialog()| now support user-defined cmdline highlighting.
```

Comment:
@bfredl asked this somewhere above: it is not guaranteed to be called, if real hooks cause any problems then I would rather add code which detects those problems and stops using hook then do anything. Currently this includes stopping calling the hook in case of arabic characters present. And handler is not called on each input character, it is called on *redraw*, minus cases when hook output can be taken from cache. I don’t have any information regarding what caused redraw, so one will have to compute a diff: hook always receives the whole drawn command-line. And whatever other uses you have in mind, note that hook is not supposed to edit input, `:echo` anything or issue any errors.